### DISCUSSION TITLE
Rating component faded out as a default appearance

### DISCUSSION THREAD

**Discussion by ffranchina** [saadeghi/daisyui#1310] [Q&A]
Hello everyone and thanks for this amazing project and community!
I decided to open a new discussion about the default appearance of the rating component when nothing is selected:
![immagine](https://user-images.githubusercontent.com/8852116/199798461-bef2444a-6e52-4dbe-a756-3d89d6cb638f.png)
Even if nothing is selected, it really looks like the default value is 5 stars.

With my current knowledge of CSS, by looking at the approach taken in developing the component, I understand the reasons of the behavior, I think. IMHO, unfortunately, it's a bit counter-intuitive but I see no options to fix it without adding extra dirty workaround.

My final question is: if you also think it's a bit off from an expected behavior, do you think it is fixable o you know a way (not too hackish) to fix it?

Thanks in advance!

**Comment by saadeghi** [comment]
There's a limitation about this.  

This is how we're currently styling the radio buttons:  
All radio buttons have opacity of `1` and very time a radio button is checked, we lower the opacity of all siblings that come after the checked radio button. This way when you pick the 3rd radio, 1 to 3 will have opacity of `1` and 4th and 5th will have the opacity of `0.2`.  
Unfortunately there's no way for CSS to select only the previous siblings so we can't by default set the opacity of all radio buttons to `0.2` and then set the opacity of checked one and all previous siblings to `1`.  

There are 2 workarounds for this issue:
1. set the opacity to `0.2` if the parent has no checked checkbox using the new CSS `:has()` selector:
  https://play.tailwindcss.com/xNdSmUy0pc?file=css
    ```css
    .rating:not(:has(input:checked)) :where(input) {
      --tw-bg-opacity: 0.2;
    }
    ```
    Warning: This solution doesn't work on Firefox yet: https://caniuse.com/css-has

2. Use [`rating-hidden` example](https://daisyui.com/components/rating/#with-rating-hidden). If you add a `rating-hidden` class for the first radio and check that first radio, it will be hidden and all visible radio buttons will have lower opacity as expected.  
  I suggest this solution because not only fixes the visual when no checkbox is selected, but also allows the user to remove their star rating by clicking the very left end part of stars.

**Reply by ffranchina** [reply]
Thanks a lot for the very clear and complete answer! It really helped me to solve my issues! :smile: 
Have a good day!
